File: Source/Microsoft.Teams.Apps.Timesheet/Models/TimesheetStatus.cs
// <copyright file="TimesheetStatus.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

namespace Microsoft.Teams.Apps.Timesheet.Models
{
    /// <summary>
    /// The enumeration that holds the status of timesheet.
    /// </summary>
    public enum TimesheetStatus
    {
        /// <summary>
        /// Represents nothing has been done in timesheet.
        /// </summary>
        None,

        /// <summary>
        /// Indicated that the timesheet saved by user.
        /// </summary>
        Saved,

        /// <summary>
        /// Represents that the timesheet submitted by user.
        /// </summary>
        Submitted,

        /// <summary>
        /// Represents that the timesheet approved by manager.
        /// </summary>
        Approved,

        /// <summary>
        /// Represents that the timesheet rejected by manager.
        /// </summary>
        Rejected,
    }
}

File: Source/Microsoft.Teams.Apps.Timesheet/ClientApp/src/models/timesheet-status.ts
// <copyright file="timesheet-status.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// The enumeration that holds the status of timesheet.
export enum TimesheetStatus {
    // Indicates that the timesheet for calendar date not yet filled.
    None,

    // Indicates that the timesheet saved by user.
    Saved,

    // Indicates that the timesheet submitted by user.
    Submitted,

    // Indicates that the timesheet approved by manager.
    Approved,

    // Indicates that the timesheet rejected by manager.
    Rejected
}